erpnext/agriculture/doctype/destroyed_plant_log/destroyed_plant_log.py

```diff
@@ -0,0 +1,22 @@
+# -*- coding: utf-8 -*-
+# Copyright (c) 2020, Frappe Technologies Pvt. Ltd. and contributors
+# For license information, please see license.txt
+
+from __future__ import unicode_literals
+import frappe
+from frappe import _
+from frappe.model.document import Document
+
+class DestroyedPlantLog(Document):
+	def on_submit(self):
+		doc = frappe.get_doc('Plant Batch',self.plant_batch)
+		if doc.untracked_count == 0:
+			frappe.throw(_("The plant batch must have an untracked count"))
+		elif int(self.destroy_count) <= 0:
+			frappe.throw(_("Destroy count cannot be less than 0"))
+		elif doc.untracked_count < int(self.destroy_count):
```

Comment:
- you don't need to typecast to `int` if the fieldtype id `int` already in the field meta.

Reply:
This type conversion is for the value coming from plant_batch.
that is of string type.